internal/databases/redis/backup_push_handler.go

```diff
@@ -0,0 +1,80 @@
+package redis
+
+import (
+	"bufio"
+	"github.com/go-redis/redis"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/internal/tracelog"
+	"github.com/wal-g/wal-g/utility"
+	"math"
+	"os"
+	"time"
+)
+
+func HandleBackupPush(uploader *Uploader) {
+	// Configure folders
+	redisDataFoler, ok := internal.GetSetting("WALG_REDIS_DATA_FOLDER")
+	if !ok {
+		redisDataFoler = "/var/lib/redis"
+	}
+	uploader.UploadingFolder = uploader.UploadingFolder.GetSubFolder(utility.BaseBackupPath)
+
+	// Get Redis connection
+	client := getRedisConnection()
+
+	// Init backup process
+	currentTime := time.Now()
+	initializeBackupSaving(client)
+
+	// Wait for new backup on disk.
+	// DISCUSS: Maybe add Timer to stop trying after some time?
+	waitForNewBackup(currentTime, redisDataFoler)
+
+	// Upload backup
+	uploadBackup(uploader, redisDataFoler)
+}
+
+func initializeBackupSaving(client *redis.Client) {
+	err := client.BgSave().Err()
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("%+v\n", err)
+	}
+}
+
+func waitForNewBackup(currentTime time.Time, redisDataFoler string) {
+	for {
+		if _, err := os.Stat(redisDataFoler + "/dump.rdb"); !os.IsNotExist(err) {
+			break
+		}
+	}
+
+	for {
+		stat, err := os.Stat(redisDataFoler + "/dump.rdb")
+		if err != nil {
+			tracelog.ErrorLogger.Fatalf("%+v\n", err)
+		}
+
+		if math.Abs(currentTime.Sub(stat.ModTime()).Seconds()) < 10 {
+			tracelog.InfoLogger.Println("Backup has been made not so long ago, not going to wait for BGSAVE")
+			break
+		}
+
+		if stat.ModTime().After(currentTime) {
+			break
+		}
+	}
+}
+
+func uploadBackup(uploader *Uploader, redisDataFoler string) {
+	f, err := os.Open(redisDataFoler + "/dump.rdb")
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("%+v\n", err)
+	}
+	defer utility.LoggedClose(f, "")
+	reader := bufio.NewReader(f)
+	namedReader := internal.NewNamedReaderImpl(reader, "base_"+time.Now().Format(time.RFC3339)) // DISCUSS: Is it correct backup name?
```

Comment:
We prefer to name backups, thinking of DB inner structure. For PG we use LSN - logical sequence number (logical time for DB). For MySQL, we have to use ordinary time just because we can't use anything else. So, if for Redis there is some simple analog for LSN, then let's use it in backup names instead.

Reply:
As far as i'm concerned, there is no analog of LSN in Redis, I suppose it is connected with overall simplicity of Redis as plain key-value storage. 
By the way, is RFC3339 format appropriate for backup suffix, we don't need to have naming with precision to nanoseconds, right?

Reply:
Not so long ago @g0djan resolved this issue with the precision of time. Let's wait, what he will tell here.